core/data/resolve.go

```diff
@@ -125,14 +125,23 @@ type ResolutionDetails struct {
 func GetResolutionDetails(toResolve string) (*ResolutionDetails, error) {
 
 	//todo optimize, maybe tokenize first
+	details := &ResolutionDetails{}
 
-	dotIdx := strings.Index(toResolve, ".")
+	bracketIdx := strings.Index(toResolve, "]")
+	exprLen := len(toResolve)
+	if bracketIdx == (exprLen-1) && (strings.HasPrefix(toResolve, "property") || strings.HasPrefix(toResolve, "env")) {
+		//$property[] or $env[] resolution
+		itemIdx := strings.Index(toResolve, "[")
+		details.ResolverName = toResolve[:itemIdx]
+		details.Property = toResolve[itemIdx+1 : exprLen-1]
+		return details, nil
```

Comment:
have issue to handle the array field case: `$property[propertyName].fieldname.array[0]`

Reply:
@lixingwang : Currently, we support only primitive types. As use cases evolve, we will add support for Object and Array in future.